include/mrdox/Metadata/Template.hpp

```diff
@@ -14,39 +14,254 @@
 #define MRDOX_METADATA_TEMPLATE_HPP
 
 #include <mrdox/Platform.hpp>
-#include <mrdox/Metadata/TemplateArg.hpp>
-#include <mrdox/Metadata/TemplateParam.hpp>
+#include <mrdox/ADT/Optional.hpp>
+#include <mrdox/Metadata/Type.hpp>
+#include <optional>
+#include <string>
+#include <vector>
 
 namespace clang {
 namespace mrdox {
 
+enum class TParamKind : int
+{
+    None = 0, // for bitstream
+    // template type parameter, e.g. "typename T" or "class T"
+    Type,
+    // template non-type parameter, e.g. "int N" or "auto N"
+    NonType,
+    // template template parameter, e.g. "template<typename> typename T"
+    Template,
```

Comment:
Out of Type, NonType, or Template, figure out which one is statistically more likely to appear and make that the "zero" value.